### DISCUSSION TITLE
Error tool tips lose styling when using overflowWidgetsDomNode

### DISCUSSION THREAD

**Discussion by parkertomatoes** [microsoft/monaco-editor#3817] [Q&A]
I was trying to solve the issue of clipped tool-tips when embedding an editor in a container with an `overflow: hidden` style.

This code reproduces _that_ problem in the playground. Hover over the error squiggles to observe.
```javascript
var jsCode = [
	'this is an error'
].join('\n');

const container = document.getElementById("container");
const subContainer = document.createElement('div');
container.appendChild(subContainer);

subContainer.style.marginTop = '100px';
subContainer.style.overflow = 'hidden';
subContainer.style.height = '200px';

var editor = monaco.editor.create(subContainer, {
	value: 'this is an error',
	language: "javascript"
});
```

I tried to fix it by using the `overflowWidgesDomNode` option:
```javascript
var editor = monaco.editor.create(subContainer, {
	value: 'this is an error',
	language: "javascript",
        overflowWidgetsDomNode: document.body,
        fixedOverflowWidgets: true
});
```

Which worked, but all the styling disappears from the tool-tip:
![image](https://user-images.githubusercontent.com/6298931/100244171-a2a48e00-2f04-11eb-8b6d-b1f86acc6d9b.png)

I traced the issue to here:
https://github.com/microsoft/vscode/blob/master/src/vs/editor/contrib/hover/hover.ts#L339

Where a bunch of styles are added under `.monaco-editor .monaco-hover`. Adding this code to the playground code fixes the colors (though "peek problem") is still unstyled:
```javascript
for (const styleElement of document.getElementsByTagName('style')) {
    styleElement.innerHTML = styleElement.innerHTML.replaceAll('.monaco-editor .monaco-hover', '.monaco-hover');
}
```

Is a bug, or am I using this feature incorrectly?

Thanks,

**Comment by blutorange** [comment]
Workaround for the CSS issue seems to be to add the `monaco-editor` style class to the container for the `overflowWidgetsDomNode` (see issue #2156)

```javascript
document.querySelector('body').classList.add('monaco-editor');
```

But `overflowWidgetsDomNode` also seems to be having issues with events (e.g. renaming via F2 does not work), so this option probably shouldn't be used for now.

**Comment by kristw** [comment]
Thanks for the workaround!
